plugins/inputs/t128_metrics/t128_metrics.go

```diff
@@ -0,0 +1,269 @@
+package t128_metrics
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// T128Metrics is an input for metrics of a 128T router instance
+type T128Metrics struct {
+	BaseURL                 string             `toml:"base_url"`
```

Comment:
It feels a little odd to have the struct that is holding the http client and stuff to also hold the passed in config, but I don't see anything strictly wrong with it. From a quick look, it looks like this is common input practice though so 🤷

Reply:
Yeah, it felt strange to me at first too, but it is how I see it done in other inputs. There shouldn't be anything wrong with the practice.